This patch is based on the commit described below, taken from the upstream time64
pull request and adapted for use in the Debian package by Peter Michael Green.

commit d8a1d16f489c0b2d7064eb2863378f7e45a696c3
Author: Ola x Nilsson <olani@axis.com>
Date:   Mon Mar 27 14:30:10 2023 +0200

    linux: Set RLIM_INFINITY for mips GNU libc with _FILE_OFFSET_BITS=64

Index: libc/src/unix/linux_like/linux/arch/mips/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/arch/mips/mod.rs
+++ libc/src/unix/linux_like/linux/arch/mips/mod.rs
@@ -364,10 +364,12 @@ cfg_if! {
 }
 
 cfg_if! {
-    if #[cfg(
+    if #[cfg(all(gnu_time64_abi, any(target_arch = "mips", target_arch = "mips32r6")))] {
+        pub const RLIM_INFINITY: ::rlim_t = !0;
+    } else if #[cfg(all(
         any(target_arch = "mips", target_arch = "mips32r6"),
-        any(target_env = "gnu", target_env = "uclibc")
-    )] {
+        target_env = "uclibc"
+    ))] {
         pub const RLIM_INFINITY: crate::rlim_t = 0x7fffffff;
     }
 }
